web/server.go

```diff
@@ -0,0 +1,67 @@
+package web
+
+import (
+	"context"
+	"net/http"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+	"time"
+)
+
+// Server will create a http.Server from the Go standard library
+type Server struct {
+	Addr    string        // TCP address to listen on, ":http" if empty
+	Handler http.Handler  // handler to invoke, http.DefaultServeMux if nil
+	Timeout time.Duration // Timeout to wait for shutdown
+}
+
+// ListenAndServe mirrors the function from the Go standard library
+// ListenAndServe always returns a non-nil error
+func (s *Server) ListenAndServe() error {
+	server := http.Server{
+		Addr:    s.Addr,
+		Handler: s.Handler,
+	}
+
+	// We want to report the listener is closed.
+	var wg sync.WaitGroup
+	wg.Add(1)
+
+	done := make(chan struct{})
+	var output error
+	// Start the listener.
+	go func() {
+		output = server.ListenAndServe()
+		close(done)
+		wg.Done()
+	}()
+
+	// Listen for an interrupt signal from the OS. Use a buffered
+	// channel because of how the signal package is implemented.
+	osSignals := make(chan os.Signal, 1)
+	signal.Notify(osSignals, os.Interrupt, syscall.SIGTERM)
+
+	// Wait for a signal to shutdown.
```

Comment:
Actually I would favor slightly more information here in the comments, i.e. a bit more information why there are 2 signals and why we need to wait on both

Reply:
*signals -> channels
(Sorry I don't know how to edit comment on mobile grrr)

Reply:
Ok, I'll add it in.